scripts/contentScript.js

```diff
@@ -666,7 +808,62 @@ const orchestrator = (tabUrl, websiteUrl, userId) => {
 				}
 			}
 		}
-	})
+
+		if (urlObj[2] === 'gitlab.com') {
+
+			if(urlObj.length === 4 && !urlObj[3]){
+				//TODO: need repo_owner from userID
+								
+
+			}
+
+			else if (urlObj.length > 4){	
+
+				const ownerName = urlObj[3];
+				const repoName = urlObj[4];
+				showFloatingActionButton(ownerName, repoName, userId, websiteUrl, 'gitlab');
+				//showing relevant MRs
+				if (urlObj[5] && (urlObj[6] === 'merge_requests') && !urlObj[7]){					
+
+					const body = {
+						"repo_owner": ownerName,
+						"repo_name": repoName,
+						"user_id": userId,
+						"repo_provider": Provider.GITLAB
+					}
+
+					const url = `${backendUrl}/relevance/pr`;
+					const highlightedMRIds = await apiCall(url, body); 
+					highlightRelevantMRs(highlightedMRIds);
+				}
+				//showing a particular MR
+				if(urlObj[5] && (urlObj[6] === 'merge_requests') && urlObj[7])
+				{
+					const prNumber = parseInt(urlObj[7]);
+					const body = {
+						"repo_owner": ownerName,
+						"repo_name": repoName,
+						"user_id": userId,
+						"pr_number": prNumber,
+						"repo_provider": Provider.GITLAB
+					}
+					const url = `${backendUrl}/relevance/pr/files`;
+					const response = await apiCall(url, body); 
+					showImpFileInMR(response); 
+					const hunk_info_body = {
+						"repo_owner": ownerName,
+						"repo_name": repoName,
+						"user_id": userId,
+						"pr_number": prNumber,
+						"repo_provider": Provider.GITLAB
+					}
+					const hunk_info_url = `${backendUrl}/relevance/hunkinfo`; 
+					const hunk_info_response = await apiCall(hunk_info_url, hunk_info_body);
+					gitlabHunkHighlight(hunk_info_response);
```

Comment:
Why repeat this pattern for GitLab when we are writing new code?
This was a tech debt we took in GitHub because we first launched without the hunk-level highlighting and then launched for hunk-level separately. If you don't need two API calls, please don't make two API calls.

It would also be great if you can fix the issue for GitHub. We are unnecessarily making two API calls when there is only a need for one. @tapishr Please look into this - let us know how feasible is it and how much time will it take.

Reply:
LOGAF: 7/10

Reply:
I will have to rewrite the relevance api for that. Would doing that in a separate pr be better?